boa/src/syntax/ast/token.rs

```diff
@@ -42,6 +42,17 @@ impl Debug for VecToken {
     }
 }
 
+/// Represents the type differenct types of numeric literals.
+#[cfg_attr(feature = "serde-ast", derive(Serialize, Deserialize))]
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum NumericLiteral {
+    /// A floating point number
+    Number(f64),
```

Comment:
I would probably change this `Number` for `Real` or something like that, to differentiate from an `Integer`.

Reply:
In V8 they have `Number` and `Smi`, so should I do it like V8 or `Real` and `Integer`?

Reply:
> In V8 they have `Number` and `Smi`, so should I do it like V8 or `Real` and `Integer`?

I think it's more accurate if we call `Real` or even `Rational` to non-integers.